CONTRIBUTING.rst

```diff
@@ -166,9 +188,13 @@ Before you submit a pull request, check that it meets these guidelines:
 
 #. The pull request should include tests and should aim to provide `code coverage <https://en.wikipedia.org/wiki/Code_coverage>`_ for all new lines of code. You can use the `--cov-report html --cov xhydro` flags during the call to ``pytest`` to generate an HTML report and analyse the current test coverage.
 
-#. If the pull request adds functionality, the docs should also be updated. Put your new functionality into a function with a docstring, and add the feature to the list in ``README.rst``.
+#. All functions should be documented with `docstrings` following the `numpydoc <https://numpydoc.readthedocs.io/en/latest/format.html>`_ format.
+
+#. If the pull request adds functionality, either update the documentation or create a new notebook in the `docs/notebooks` directory that demonstrates the feature. the Jupyter notebooks located in the documentation should also be updated. Library-defining features should also be listed in ``README.rst``.
+
+#. The ChangeLog should be updated with a brief description of the changes made in the pull request. If this is your first contribution to the project, please add your name to the `AUTHORS.rst` and `.zenodo.json` files.
 
-#. The pull request should work for Python 3.9, 3.10, 3.11, and 3.12. Check that the tests pass for all supported Python versions.
+#. The pull request should work for all currently supported Python versions. Check the `pyproject.toml` or `tox.ini` files for the supported versions.
```

Comment:
This is better than the current writing. Definitely to be pushed up to the template.